weaviate/collections/tenants.py

```diff
@@ -171,9 +175,26 @@ def __get_with_grpc(
             for tenant in response.tenants
         }
 
-    def __map_input_tenant(self, tenant: TenantInputType) -> TenantInput:
+    def __map_create_tenant(self, tenant: TenantCreateInputType) -> TenantCreate:
         if isinstance(tenant, str):
-            return TenantInput(name=tenant)
+            return TenantCreate(name=tenant)
+        if isinstance(tenant, Tenant):
+            if tenant.activity_status in [
+                TenantActivityStatus.FREEZING,
+                TenantActivityStatus.UNFROZEN,
+                TenantActivityStatus.UNFREEZING,
+                TenantActivityStatus.FROZEN,
+            ]:
+                raise WeaviateInvalidInputError(
+                    f"Tenant activity status must be either 'HOT' or 'COLD'. Other statuses are read-only and cannot be set. Tenant: {tenant.name} had status: {tenant.activity_status}"
+                )
```

Comment:
should we do `tenant.activity_status not in HOT, COLD` ? In case there are more in the future?

Reply:
Ah yes, that's a much better idea!